spynnaker/pyNN/models/common/neuron_recorder.py

```diff
@@ -1,19 +1,107 @@
 import logging
+import numpy
+
+from data_specification.enums import DataType
+from spinn_utilities.progress_bar import ProgressBar
 from spynnaker.pyNN.models.common import recording_utils
-from .abstract_uint32_recorder import AbstractUInt32Recorder
 from spinn_front_end_common.utilities import exceptions as fec_excceptions
+from spinn_front_end_common.utilities import globals_variables
 
 logger = logging.getLogger(__name__)
 
 
-class NeuronRecorder(AbstractUInt32Recorder):
+class NeuronRecorder(object):
+    N_BYTES_PER_NEURON = 4
+    N_CPU_CYCLES_PER_NEURON = 4
 
     def __init__(self, allowed_variables):
-        AbstractUInt32Recorder.__init__(self)
         self._record = dict()
         for variable in allowed_variables:
             self._record[variable] = False
 
+    def n_neurons_recording(self, variable, vertex_slice):
+        return vertex_slice.n_atoms
+
+    def neurons_recording(self, variable, vertex_slice):
+        return range(vertex_slice.lo_atom, vertex_slice.hi_atom + 1)
+
+    def get_neuron_sampling_interval(self, variable):
+        """
+        Returns the current sampling interval for this variable
+        :param variable: PyNN name of the variable
+        :return: Sampling interval in micro seconds
+        """
+        return globals_variables.get_simulator().machine_time_step
+
+    def get_matrix_data(
+            self, label, buffer_manager, region, placements, graph_mapper,
+            application_vertex, variable, n_machine_time_steps):
+        """ method for reading a uint32 mapped to time and neuron ids from\
+        the SpiNNaker machine
+
+        :param label: vertex label
+        :param buffer_manager: the manager for buffered data
+        :param region: the dsg region id used for this data
+        :param placements: the placements object
+        :param graph_mapper: the mapping between application and machine\
+            vertices
+        :param application_vertex:
+        :param variable: PyNN name for the variable (V, gsy_ihn ect
+        :type variable: str
+        :param n_machine_time_steps:
+        :return:
+        """
+        vertices = graph_mapper.get_machine_vertices(application_vertex)
+        progress = ProgressBar(
+                vertices, "Getting {} for {}".format(variable, label))
+        sampling_interval = self.get_sampling_interval(variable)
```

Comment:
I don't think this is a thing - there is self.get_neuron_sampling_interval...

Reply:
fixed